### PR TITLE
gpuav: Handle VkBufferUsageFlags2CreateInfo

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Caught in the Granite `multi-draw-indirect` test

Self validation was mad because we never handled `VkBufferUsageFlags2CreateInfo` correctly

### CODE REVIEW

File: tests/unit/gpu_av_descriptor_class_general_buffer.cpp [hunk 1]
```diff
@@ -2527,6 +2527,33 @@ TEST_F(NegativeGpuAVDescriptorClassGeneralBuffer, LoopsWithBranch) {
     ComputeStorageBufferTest(cs_source, true, 256, "VUID-vkCmdDispatch-storageBuffers-06936");
 }
 
+TEST_F(NegativeGpuAVDescriptorClassGeneralBuffer, ComplexCalculate) {
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_descriptor_class_general_buffer.cpp:2530]: unrelated, but another random good edge case test wanted to add, so snuck in this PR :shrug:  (#SaveCI)

**Review by artem-lunarg — APPROVED** [review]

**Review by arno-lunarg — APPROVED** [review]
